Tests/SparkConnectTests/DataFrameTests.swift

```diff
@@ -744,6 +744,22 @@ struct DataFrameTests {
     await spark.stop()
   }
 
+  @Test
+  func withWatermark() async throws {
+    let spark = try await SparkSession.builder.getOrCreate()
+    let df = try await spark
+      .sql("""
+        SELECT * FROM VALUES
+          (1, now()),
+          (1, now() - INTERVAL 1 HOUR),
+          (1, now() - INTERVAL 2 HOUR)
+          T(data, eventTime)
+        """)
+      .withWatermark("eventTime", "1 minute") // This tests only API for now
```

Comment:
I manually checked from the UI.

<img width="836" alt="Screenshot 2025-05-15 at 21 59 26" src="https://github.com/user-attachments/assets/d445a951-4442-4dee-86b3-4c61858e3fa2" />